test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -33,55 +39,256 @@ using namespace dev::solidity::test;
 using namespace std;
 using namespace soltest;
 
-dev::solidity::test::ParameterList ContractABIUtils::parametersFromJson(
+namespace
+{
+
+using ParameterList = dev::solidity::test::ParameterList;
+
+auto arraySize(string const& _arrayType) -> size_t
+{
+	auto leftBrack = _arrayType.find("[");
+	auto rightBrack = _arrayType.find("]");
+
+	soltestAssert(leftBrack != string::npos && rightBrack != string::npos, "");
+
+	string size = _arrayType.substr(leftBrack + 1, rightBrack - leftBrack - 1);
+	return static_cast<size_t>(stoi(size));
+}
+
+auto isBool(string _type) -> bool
+{
+	return regex_match(_type, regex{"(bool)"});
+}
+
+auto isUint(string _type) -> bool
+{
+	return regex_match(_type, regex{"(uint\\d*)"});
```

Comment:
Please check that this does not match on `someuintabc`

Reply:
It does not match `someuintabc`, but I removed the subexpressions.